core/src/main/scala/org/apache/spark/scheduler/Pool.scala

```diff
@@ -55,6 +55,8 @@ private[spark] class Pool(
         new FairSchedulingAlgorithm()
       case SchedulingMode.FIFO =>
         new FIFOSchedulingAlgorithm()
+      case _ =>
+        throw new RuntimeException(s"The scheduler mode $schedulingMode is not supported by Spark.")
```

Comment:
In [TaskSchedulerImpl](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala#L118) they use: "Unrecognized spark.scheduler.mode: $schedulingModeConf" so I'd change it to "Unsupported spark.scheduler.mode: $schedulingMode".

I'm also not sure about the other places like https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala#L133-L138. I know that the change would not let it be executed with `NONE` or unsupported scheduling modes, but when Pool changes...just thinking aloud.